src/definition/populate_test.go

```diff
@@ -80,6 +80,76 @@ func metricsNamingManipulator(ms metric.MetricSet, entity sdk.Entity, clusterNam
 	return ms.SetMetric("displayName", entity.Name, metric.ATTRIBUTE)
 }
 
+func TestIntegrationProtocol2PopulateFunc_AddingNewMetricSetToEntity(t *testing.T) {
+	rawGroupsSample1 := RawGroups{
+		"test": {
+			"entity_id_1": RawMetrics{
+				"raw_metric_name_1": 1,
+			},
+		},
+	}
+	rawGroupsSample2 := RawGroups{
+		"test": {
+			"entity_id_1": RawMetrics{
+				"raw_metric_name_2": "foo",
+			},
+		},
+	}
+	specs := SpecGroups{
+		"test": SpecGroup{
+			TypeGenerator: fromGroupEntityTypeGuessFunc,
+			Specs: []Spec{
+				{"metric_1", FromRaw("raw_metric_name_1"), metric.GAUGE, true},
+				{"metric_2", FromRaw("raw_metric_name_2"), metric.ATTRIBUTE, true},
+			},
+		},
+	}
+	integration, err := sdk.NewIntegrationProtocol2("nr.test", "1.0.0", new(struct{}))
+	if err != nil {
+		t.Fatal()
+	}
+
+	expectedEntityData, err := sdk.NewEntityData("entity_id_1", "playground:test")
+	if err != nil {
+		t.Fatal()
+	}
+
+	expectedMetricSet := metric.MetricSet{
+		"event_type":  "TestSample",
+		"metric_1":    1,
+		"entityName":  "playground:test:entity_id_1",
+		"displayName": "entity_id_1",
+		"clusterName": "playground",
+	}
+	expectedEntityData.Metrics = []metric.MetricSet{expectedMetricSet}
+
+	populated, errs := IntegrationProtocol2PopulateFunc(
+		integration,
+		defaultNS,
+		&version.Info{GitVersion: "v1.15.42"},
+		fromGroupMetricSetTypeGuessFunc,
+		metricsNamingManipulator,
+		clusterMetricsManipulator,
+	)(rawGroupsSample1, specs)
+	assert.True(t, populated)
+	assert.Empty(t, errs)
+	assert.Contains(t, integration.Data, &expectedEntityData)
```

Comment:
It seems to me those asserts should be part of other test. We should only test the behavior of multiple metricsets here I think.

So those asserts are perhaps duplicated in another test.

This will also make test easier, as we won't need to further mutate `expectedMetricSet`.

Reply:
you're right i will remove this , there is no point to double test this.